src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PrivateDataRetrievalService.java

```diff
@@ -103,12 +113,16 @@ private PrivateBoDataListApi getBeneficialOwnersData(String companyNumber)
         ApiLogger
             .info(String.format("Retrieved %d Beneficial Owners for Company Number %s",
                 numberOfBOs, companyNumber));
-      } else {
-        ApiLogger.info("No Beneficial Owners found for Company Number " + companyNumber);
       }
 
       return beneficialOwnerDataList;
-
+    } catch (ApiErrorResponseException e) {
+      if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+        ApiLogger.info(String.format("No Beneficial Owners found for Company Number %s", companyNumber), logMap);
+        var beneficialOwnerDataList = new PrivateBoDataListApi(Collections.emptyList());
+        return beneficialOwnerDataList;
+      }
```

Comment:
Need to log the error here before throwing. That's what we do for 127-129 general error path.